pkg/drawio/subnetLayoutOverlap.go

```diff
@@ -0,0 +1,314 @@
+package drawio
+
+// subnetLayoutOverlap is for handling overlaps in subnet mode.
+// it handle two kind of issues:
+// 1. src and dst of a line are intersect - this is solved by adding a point to the line.
+//    the point should be outside both src and dst
+// 2. two lines are in opposite direction - this is solved by changing the the exit point of the line from the src
+
+type subnetLayoutOverlap struct {
+	xIndexes map[*col]int
+	yIndexes map[*row]int
+	network  SquareTreeNodeInterface
+}
+
+func newSubnetLayoutOverlap(network SquareTreeNodeInterface, m *layoutMatrix) *subnetLayoutOverlap {
+	lyO := subnetLayoutOverlap{xIndexes: map[*col]int{}, yIndexes: map[*row]int{}, network: network}
+	lyO.setIndexes(m)
+	return &lyO
+}
+
+// some of the calculations are done on a matrix that take in account only the thick rows and cols
+// (basically ignores the border rows/cols)
+// so, as first step, we indexing all the thick rows/cols (set lyO.xIndexes and lyO.yIndexes)
+// later, to find the location of a square in this matrix, we take their indexes from lyO.xIndexes and lyO.yIndexes
```

Comment:
Do we assume that for every line the src/dst col/row are always thick? If so, add a comment here.